libass/ass.c

```diff
@@ -1235,6 +1235,31 @@ static char *sub_recode(ASS_Library *library, char *data, size_t size,
 }
 #endif                          // ICONV
 
+FILE* ass_fopen(const char* _FileName, const char* _Mode) {
+#if defined(_WIN32) && !defined(__CYGWIN__)
+    wchar_t* wpath, * wmode;
+    UINT cur_cp = CP_UTF8;
```

Comment:
I think we've already decided that `MultiByteToWideChar` with `CP_UTF8` is not enough: we need WTF-8 to ensure safe round-trip of any wide sequence, even broken UTF-16.

Reply:
[here](https://github.com/libass/libass/pull/526#issuecomment-873599277) referenced that MB_ERR_INVALID_CHARS is needed. So back the flag to 0 ?

Reply:
AFAIK, `MultiByteToWideChar` cannot do the thing at all, so a custom conversion function is needed.

Reply:
just like sub_recode?

Reply:
'sub_recode()` uses iconv, which doesn't support WTF-8 too.